crates/solidity/outputs/cargo/tests/src/doc_examples/using_queries.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::Path;
+
+use anyhow::Result;
+use infra_utils::paths::PathExtensions;
+use semver::Version;
+use slang_solidity::kinds::RuleKind;
+use slang_solidity::language::Language;
+use slang_solidity::parse_output::ParseOutput;
+use slang_solidity::query::Query;
+
+fn parse_doc_input_file<T: AsRef<Path>>(path: T) -> Result<ParseOutput> {
+    let input_path =
+        Path::repo_path(Path::new("documentation/public/user-guide/inputs").join(path.as_ref()));
+
+    let input_path = input_path.unwrap_str();
+
+    let source = std::fs::read_to_string(input_path)?;
+    let source = source.trim();
+
+    let language = Language::new(Version::new(0, 8, 0))?;
+
+    Ok(language.parse(RuleKind::SourceUnit, source))
+}
+
+#[test]
+fn using_queries() -> Result<()> {
+    #[allow(unused_variables, clippy::items_after_statements)]
+    {
+        let parse_output = parse_doc_input_file("using-the-cursor.sol")?;
+        // --8<-- [start:creating-a-query]
+        use slang_solidity::query::Query;
+
+        // Any `Cursor` can be used to create a query.
+        let cursor = parse_output.create_tree_cursor();
+
+        let query = Query::parse("[ContractDefinition]").unwrap();
+        let result /* : QueryResultIterator */ = cursor.query(vec![query]);
```

Comment:
I suggest uncommenting it, to type-check it during the build:

```suggestion
        let result: QueryResultIterator = cursor.query(vec![query]);
```

Reply:
Done, but I hope this doesn't hint to the user that they need to use the explicit signature to help the type system with narrowing; this should be purely optional.

Reply:
Valid point. But IMO, the type checking (so that docs don't go out of date) outweight that concern.
We had a few instances of this hapenning before we started running doc examples as unit tests.